File: solutions/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i, num in enumerate(nums):
            diff = target - num
            for j, num2 in enumerate(nums[i+1:]):
                if num2 == diff:
                    return [i, j+i+1]
                    

File: solutions/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int m,n;
        
        //creating and inserting values in hashmap
        unordered_map<int,int> map;
        for(auto x: nums) map[x]++;
        
        //to get the first element index
        for(int i = 0; i<nums.size() ; i++)
        {
            map[nums[i]]--;
            if(map[nums[i]]==0) map.erase(nums[i]);
            if(map.find(target-nums[i])!=map.end())
            {
                m = i;
                break;
            }
        }
        //to get second element index
        for(int i = 0; i<nums.size() ; i++)
        {
            if(nums[i] + nums[m]==target && m!=i)
            {
                n = i;
                break;
            }
        }
        return {m,n};
    }
};
